CONTRIBUTING.md

```diff
@@ -2,4 +2,159 @@
 
 First of all, thank you for contributing to MeiliSearch! The goal of this document is to provide everything you need to know in order to contribute to MeiliSearch and its different integrations.
 
-// TODO
\ No newline at end of file
+<!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
+
+- [Assumptions](#assumptions)
+- [How to Contribute](#how-to-contribute)
+- [Development Workflow](#development-workflow)
+- [Git Guidelines](#git-guidelines)
+
+<!-- /MarkdownTOC -->
+
+## Assumptions
+
+1. **You're familiar with [GitHub](https://github.com) and the [Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)(PR) workflow.**
+2. **You've read the MeiliSearch [documentation](https://docs.meilisearch.com) and the [README](/README.md).**
+3. **You know about the [MeiliSearch community](https://docs.meilisearch.com/resources/contact.html). Please use this for help.**
+
+## How to Contribute
+
+1. Make sure that the contribution you want to make is explained or detailed in a GitHub issue! Find an [existing issue](https://github.com/meilisearch/meilisearch-java/issues/) or [open a new one](https://github.com/meilisearch/meilisearch-java/issues/new).
+2. Once done, [fork the meilisearch-java repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) in your own GitHub account. Ask a maintainer if you want your issue to be checked before making a PR.
+3. [Create a new Git branch](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository).
+4. Review the [Development Workflow](#workflow) section that describes the steps to maintain the repository.
+5. Make your changes.
+6. [Submit the branch as a PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) pointing to the `master` branch of the main meilisearch-java repository. A maintainer should comment and/or review your Pull Request within a few days. Although depending on the circumstances, it may take longer.<br>
+ We do not enforce a naming convention for the PRs, but **please use something descriptive of your changes**, having in mind that the title of your PR will be automatically added to the next [release changelog](https://github.com/meilisearch/meilisearch-java/releases/).
+
+## Development Workflow
+
+### Setup
+
+```bash
+$ ./gradlew install
+```
+
+### Tests
+
+Each PR should pass the tests to be accepted.
+
+
+```bash
+# Tests
+$ docker pull getmeili/meilisearch:latest # Fetch the latest version of MeiliSearch image from Docker Hub
+$ docker run -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --master-key=masterKey --no-analytics=true
+$ ./gradlew test
+```
+
+Integration tests will be run in your PR to check everything is OK, but you can run this test in your local environment
+
+```bash
+# Tests
+$ ./gradlew test IntegrationTest
```

Comment:
Integration guides are exluded from the basic tests, because they require a MeiliSearch instance and are a little bit long. Normal tests will just mock and test methods input/output.

Integration tests are run by default in the CI, but not in your local environment unless you ask so, by using this command

Reply:
Ok, I need to upgrade the CI so that all the tests will pass when doing a PR
let it like this for the moment 👍

Reply:
The CI runs all the tests already